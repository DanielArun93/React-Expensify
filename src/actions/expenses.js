import uuid from 'uuid';
import database from '../firebase/firebase';

//Add expenses action generators
export const addExpense = (expense) => {
    return {
        type: 'ADDEXPENSES',
        expense
    }
}

export const startAddExpense = (expenseData = {}) => {
    return (dispatch) => {
        const {
            description = '',
            note = '',
            amount = 0,
            createdAt = 0
        } = expenseData

        const expense = { description, note, amount, createdAt };
        return database.ref('expenses').push(expense).then((ref) => {
            dispatch(addExpense({
                id: ref.key,
                ...expense
            }))
        })

    }
}

//Remove expense
export const removeExpense = ({id} = {}) => {
    return {
        type: 'REMOVEEXPENSE',
        id
    }
}

//edit expense
export const editExpense = (id, update) => {
    return {
        type: 'EDITEXPENSE',
        id,
        update
    }
}