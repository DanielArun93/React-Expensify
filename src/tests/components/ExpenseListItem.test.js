import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem/ExpenseListItem';

test('should create ExpenseListItem when expense obj is provided',() => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]} />)
    expect(wrapper).toMatchSnapshot();
})