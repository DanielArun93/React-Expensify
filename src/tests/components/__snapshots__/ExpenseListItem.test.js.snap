// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create ExpenseListItem when expense obj is provided 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      Rent
    </h3>
  </Link>
  <p>
    12000
     - 
    0
  </p>
</div>
`;
